{
  "GSD": {
    "alias": "CVE-2022-40899",
    "id": "GSD-2022-40899",
    "references": [
      "https://www.suse.com/security/cve/CVE-2022-40899.html",
      "https://advisories.mageia.org/CVE-2022-40899.html",
      "https://ubuntu.com/security/CVE-2022-40899"
    ]
  },
  "gsd": {
    "metadata": {
      "exploitCode": "unknown",
      "remediation": "unknown",
      "reportConfidence": "confirmed",
      "type": "vulnerability"
    },
    "osvSchema": {
      "aliases": [
        "CVE-2022-40899"
      ],
      "details": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.",
      "id": "GSD-2022-40899",
      "modified": "2023-12-13T01:19:31.297682Z",
      "schema_version": "1.4.0"
    }
  },
  "namespaces": {
    "cve.org": {
      "CVE_data_meta": {
        "ASSIGNER": "cve@mitre.org",
        "ID": "CVE-2022-40899",
        "STATE": "PUBLIC"
      },
      "affects": {
        "vendor": {
          "vendor_data": [
            {
              "product": {
                "product_data": [
                  {
                    "product_name": "n/a",
                    "version": {
                      "version_data": [
                        {
                          "version_value": "n/a"
                        }
                      ]
                    }
                  }
                ]
              },
              "vendor_name": "n/a"
            }
          ]
        }
      },
      "data_format": "MITRE",
      "data_type": "CVE",
      "data_version": "4.0",
      "description": {
        "description_data": [
          {
            "lang": "eng",
            "value": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server."
          }
        ]
      },
      "problemtype": {
        "problemtype_data": [
          {
            "description": [
              {
                "lang": "eng",
                "value": "n/a"
              }
            ]
          }
        ]
      },
      "references": {
        "reference_data": [
          {
            "name": "https://pypi.org/project/future/",
            "refsource": "MISC",
            "url": "https://pypi.org/project/future/"
          },
          {
            "name": "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215",
            "refsource": "MISC",
            "url": "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215"
          },
          {
            "name": "https://github.com/python/cpython/pull/17157",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/pull/17157"
          },
          {
            "name": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/",
            "refsource": "MISC",
            "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
          },
          {
            "name": "https://github.com/PythonCharmers/python-future/pull/610",
            "refsource": "MISC",
            "url": "https://github.com/PythonCharmers/python-future/pull/610"
          }
        ]
      }
    },
    "gitlab.com": {
      "advisories": [
        {
          "affected_range": "<=0.18.2",
          "affected_versions": "All versions up to 0.18.2",
          "cvss_v3": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
          "cwe_ids": [
            "CWE-1035",
            "CWE-937"
          ],
          "date": "2023-01-23",
          "description": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.",
          "fixed_versions": [
            "0.18.3"
          ],
          "identifier": "CVE-2022-40899",
          "identifiers": [
            "CVE-2022-40899"
          ],
          "not_impacted": "All versions after 0.18.2",
          "package_slug": "pypi/future",
          "pubdate": "2022-12-23",
          "solution": "Upgrade to version 0.18.3 or above.",
          "title": "Denial of Service",
          "urls": [
            "https://nvd.nist.gov/vuln/detail/CVE-2022-40899",
            "https://pypi.org/project/future/",
            "https://github.com/python/cpython/pull/17157",
            "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215",
            "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
          ],
          "uuid": "59f72dd3-88c1-47e8-9757-2e68086121cf"
        }
      ]
    },
    "nvd.nist.gov": {
      "configurations": {
        "CVE_data_version": "4.0",
        "nodes": [
          {
            "children": [],
            "cpe_match": [
              {
                "cpe23Uri": "cpe:2.3:a:pythoncharmers:python-future:*:*:*:*:*:python:*:*",
                "cpe_name": [],
                "versionEndIncluding": "0.18.2",
                "vulnerable": true
              }
            ],
            "operator": "OR"
          }
        ]
      },
      "cve": {
        "CVE_data_meta": {
          "ASSIGNER": "cve@mitre.org",
          "ID": "CVE-2022-40899"
        },
        "data_format": "MITRE",
        "data_type": "CVE",
        "data_version": "4.0",
        "description": {
          "description_data": [
            {
              "lang": "en",
              "value": "An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server."
            }
          ]
        },
        "problemtype": {
          "problemtype_data": [
            {
              "description": [
                {
                  "lang": "en",
                  "value": "NVD-CWE-noinfo"
                }
              ]
            }
          ]
        },
        "references": {
          "reference_data": [
            {
              "name": "https://pypi.org/project/future/",
              "refsource": "MISC",
              "tags": [
                "Product",
                "Third Party Advisory"
              ],
              "url": "https://pypi.org/project/future/"
            },
            {
              "name": "https://github.com/python/cpython/pull/17157",
              "refsource": "MISC",
              "tags": [
                "Exploit",
                "Patch",
                "Third Party Advisory"
              ],
              "url": "https://github.com/python/cpython/pull/17157"
            },
            {
              "name": "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215",
              "refsource": "MISC",
              "tags": [
                "Third Party Advisory"
              ],
              "url": "https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215"
            },
            {
              "name": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/",
              "refsource": "MISC",
              "tags": [
                "Exploit",
                "Vendor Advisory"
              ],
              "url": "https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"
            },
            {
              "name": "https://github.com/PythonCharmers/python-future/pull/610",
              "refsource": "MISC",
              "tags": [
                "Issue Tracking",
                "Patch",
                "Third Party Advisory"
              ],
              "url": "https://github.com/PythonCharmers/python-future/pull/610"
            }
          ]
        }
      },
      "impact": {
        "baseMetricV3": {
          "cvssV3": {
            "attackComplexity": "LOW",
            "attackVector": "NETWORK",
            "availabilityImpact": "HIGH",
            "baseScore": 7.5,
            "baseSeverity": "HIGH",
            "confidentialityImpact": "NONE",
            "integrityImpact": "NONE",
            "privilegesRequired": "NONE",
            "scope": "UNCHANGED",
            "userInteraction": "NONE",
            "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
            "version": "3.1"
          },
          "exploitabilityScore": 3.9,
          "impactScore": 3.6
        }
      },
      "lastModifiedDate": "2023-01-23T18:57Z",
      "publishedDate": "2022-12-23T00:15Z"
    }
  }
}